corehq/motech/static/motech/js/connection_settings_detail.js

```diff
@@ -62,6 +63,7 @@ hqDefine("motech/js/connection_settings_detail", [
                     break;
                 case 'api_key':
                     visible = {
+                        'username': gettext("HTTP Header Name"),
```

Comment:
What is the UX in the default use case? Is this field expected to be left blank or is the default value of "Authorization" auto-populated? If blank, is the expectation obvious, or will the user likely feel compelled to enter a value?

Reply:
Good question. The default is blank and if left blank the "Authorization" will be used. I did want to make that the default in the UI but it's a bit janky due to how the fields are reused between different auth types.

Given how infrequently this is used and that if you are using API Key auth you'd almost certainly know which header to use (or know where to look it up in your service docs) I decided to leave it.

But open to push back.

Reply:
Not a great fix, but maybe a note in the field name would help? I don't love my choice of words, so feel free to change if you can think of something better.
```suggestion
                        'username': gettext("HTTP Header Name (leave empty for default 'Authorization')"),
```

Reply:
I've updated to as follows:

* clear the field value when the field label changes e.g. from `Username` to `HTTP Header`
* Set the `placeholder` for `HTTP Header` to `Authorization`